### PR TITLE
(1) Add new routine to configure for VFD SWMR: init_vfd_swmr_config()

### PR DESCRIPTION
**Pull request by vchoi-hdfgroup**
(2) Revise routine to setup fapl for VFD SWMR legacy and other integration tests: vfd_swmr_create_fapl()
(3) Update all VFD SWMR integration tests to use the above two routines
(4) Clean up VFD SWMR legacy tests: turn on compression in test script, remove #ifdef OUT H5Fflush(), message file name

### CODE REVIEW

File: test/vfd_swmr_common.c [hunk 1]
```diff
@@ -227,58 +228,78 @@ await_signal(hid_t fid)
...
+    config->pb_expansion_threshold = 0; 
+
+    config->tick_len = tick_len;
+    config->max_lag = max_lag;
+    config->writer = writer;
+    config->flush_raw_data = flush_raw_data;
+    config->md_pages_reserved = md_pages_reserved;
+
+    va_start(ap, md_file_fmtstr);
+    evsnprintf(config->md_file_path, sizeof(config->md_file_path),
```
Comment (raylu-hdf) [inline test/vfd_swmr_common.c:252]: I'm trying to understand the code here.  Does `md_file_path` need memory allocation?

Comment (vchoi-hdfgroup) [reply test/vfd_swmr_common.c:252]: Memory for H5F_vfd_swmr_config_t is allocated by the caller:

typedef struct H5F_vfd_swmr_config_t {
    int32_t     version;
    uint32_t    tick_len;
    uint32_t    max_lag;
    hbool_t     writer;
    hbool_t     flush_raw_data;
    uint32_t    md_pages_reserved;
    uint32_t    pb_expansion_threshold;
    char        md_file_path[H5F__MAX_VFD_SWMR_FILE_NAME_LEN + 1];
    char        log_file_path[H5F__MAX_VFD_SWMR_FILE_NAME_LEN + 1];
} H5F_vfd_swmr_config_t;

Comment (raylu-hdf) [reply test/vfd_swmr_common.c:252]: I see.  Thanks.

**Review by raylu-hdf — APPROVED** [review]
Looks good to me.

**Pull request merged**